pymodbus/client/base.py

```diff
@@ -78,7 +78,7 @@ def __init__(
         """Initialize a client instance."""
 
         # get optional parameters or set defaults
-        self.framer = framer(modbus_decoder())
+        self.framer = framer(modbus_decoder(), self)
```

Comment:
That change causes problems. Where did you see the problem with the code as it is.

ClientDecoder does not take an extra argument.

Reply:
I am having issues with the latest dev branch. Can't get anything working and these changes are what that made it working. I am not sure of your setup but I would request you use a new venv and try.

Reply:
A new venv have nothing to do with this!!! you can see that all tests run ok, as well as pylint etc. so do not tell me that nothing is working. Please remember that CI makes a new venv every time.

Apart from that, you should look deeper into your changes. You e.g. removed a .stop in the example, but the real problem is in the library (solved on dev_client).

For sure there are momentary problems, because I focus on the changes, while securing that I have a GREEN CI before merging anything, but the examples are a bit later, when the client is stabilised.

I suggest you use an older dev.

Reply:
```
/Users/sanjay/.virtualenvs/pymodbus/bin/python /Users/sanjay/repo/pymodbus/examples/client_sync_basic_calls.py --com tcp --framer socket --port 5020
22:35:30 INFO  client_sync:47 ### Create client object
22:35:30 INFO  client_sync_basic_calls:16 ### Reading Coil
Traceback (most recent call last):
  File "/Users/sanjay/repo/pymodbus/examples/client_sync_basic_calls.py", line 119, in <module>
    run_sync_client(demonstrate_calls)
  File "/Users/sanjay/repo/pymodbus/examples/client_sync.py", line 124, in run_sync_client
    modbus_calls(client)
  File "/Users/sanjay/repo/pymodbus/examples/client_sync_basic_calls.py", line 112, in demonstrate_calls
    handle_coils(client)
  File "/Users/sanjay/repo/pymodbus/examples/client_sync_basic_calls.py", line 17, in handle_coils
    rr = client.read_coils(1, 1, unit=UNIT)
  File "/Users/sanjay/repo/pymodbus/pymodbus/client/mixin.py", line 103, in read_coils
    return self.execute(request)  # pylint: disable=no-member
  File "/Users/sanjay/repo/pymodbus/pymodbus/client/base.py", line 178, in execute
    return self.transaction.execute(request)
  File "/Users/sanjay/repo/pymodbus/pymodbus/transaction.py", line 165, in execute
    response, last_exception = self._transact(
  File "/Users/sanjay/repo/pymodbus/pymodbus/transaction.py", line 289, in _transact
    size = self._send(packet)
  File "/Users/sanjay/repo/pymodbus/pymodbus/transaction.py", line 341, in _send
    return self.client.framer.sendPacket(packet)
  File "/Users/sanjay/repo/pymodbus/pymodbus/framer/__init__.py", line 43, in sendPacket
    return self.client.send(message)  # pylint: disable=no-member
AttributeError: 'NoneType' object has no attribute 'send'
```

Reply:
As I said before, you are using dev, which right now is unstable...if you want something stable, use an older version.

Right now you using my time to answer your messages instead of getting the problems solved.

Reply:
I am having my problems solved on my own.

Reply:
> I am open for changes, look at PR I have reviewed and merged from others!
> 
> But when you sidestep (like removing .stop) instead of solving the problem, that is not very helpful. And hopefully something you would also note if you saw me do the same.

I am reiterating it again, I do not want to step on your refactoring and thats the reason I am having these changes done at the example level for the time being. We can revert examples easily. Consider this as a testing I am doing.